backend/src/routes/auth.routes.ts

```diff
@@ -0,0 +1,18 @@
+import { Router } from 'express';
+import { AuthController } from '../controllers/auth.controller';
+import passport from 'passport';
+import { isAuthenticated } from '../middlewares/auth/authenticated.middleware';
+import { isNotAuthenticated } from '../middlewares/auth/notAuthenticated.middleware';
+import { loginValidator } from '../validators/auth/login.validator';
+import { registerValidator } from '../validators/auth/register.validator';
+import { canRegister } from '../middlewares/auth/register.middleware';
+
+const router = Router();
+const authController = new AuthController();
+
+router.post('/login', isNotAuthenticated, loginValidator, passport.authenticate("local"), authController.login);
+router.post('/register', registerValidator, canRegister, authController.register);
+router.get('/user', isAuthenticated, authController.user);
+router.delete('/logout', isAuthenticated, authController.logout);
```

Comment:
Cualquier tipo de cambio/solicitud en la autenticación deberían ser enviadas desde una método post, es una convención pero al mismo tiempo, es porque se manejan datos sencibles.
```suggestion
router.post('/logout', isAuthenticated, authController.logout);
```